server/raft.go

```diff
@@ -1101,6 +1105,25 @@ func (n *raft) ResumeApply() {
 	}
 }
 
+// DrainAndReplaySnapshot will drain the apply queue and replay the snapshot.
+// Our highest known commit will be preserved by pausing applies. The caller
+// should make sure to call ResumeApply() when handling the snapshot from the
+// queue, which will populate the rest of the committed entries in the queue.
+func (n *raft) DrainAndReplaySnapshot() bool {
+	n.Lock()
+	defer n.Unlock()
+	n.warn("Draining and replaying snapshot")
+	snap, err := n.loadLastSnapshot()
```

Comment:
Followup to my comment above, if we can not load the snapshot for some reason will we just spin?